### PR TITLE
Cleaned up error codes based off feedback in PR #2077.

### LINKED ISSUE #1044: Attestation: OE_VERIFY_FAILED was overloaded for many validation failures, need define new error codes for specific failure condition
The OE_VERIFY_FAILURE code is returned for any validation failures in the the code path of the oe_verify_report() API, which does not help narrow down the root causes, even thought the OE SDK has the exact error code for most the failure scenario. Unfortunately, the original failures were swallowed and replaced with generic  OE_VERIFY_FAILURE  code, which makes debugging attestation issues very difficult.

We need to map to original error codes to unique OE_ error code.

### PR DESCRIPTION
**Pull request by jazzybluesea**
Code review changes from previous PR #2077.
- Fixed typo: From OE_VERIFY_FAILED_OES_CMAC_MISMATCH to OE_VERIFY_FAILED_AES_CMAC_MISMATCH.
- Renamed OE_QUOTE_ENCLAVE_IDENTITY_MRSIGNER_MISMATCH  to OE_QUOTE_ENCLAVE_IDENTITY_UNIQUEID_MISMATCH.
- Renamed QE_QUOTE_ENCLAVE_IDENTITY_ISV_PROD_ID_MISMATCH to QE_QUOTE_ENCLAVE_IDENTITY_PRODUCTID_MISMATCH.
- Removed SGX specific error code QE_QUOTE_ENCLAVE_IDENTITY_ISVSVN_UNEXPECTED.
- Removed unnecessary error check.

Fixes #1044

### CODE REVIEW

File: include/openenclave/bits/result.h [hunk 1]
```diff
@@ -300,24 +300,19 @@ typedef enum _oe_result
     OE_QUOTE_ENCLAVE_IDENTITY_VERIFICATION_FAILED,
 
     /**
-     * QE identify mrsigner do not match.
+     * QE identify unique Id do not match.
```
Comment (CodeMonkeyLeet) [inline include/openenclave/bits/result.h]: Copyedit: Clean up the grammar and spelling for the comments e.g.  "identify" should be "identity". I would suggest:

- Failed to verify the identity of the quoting enclave.
- Unique ID of the quoting enclave does not match the expected value.
- Product ID of the quoting enclave does not match the expected value.
- AES CMAC of the report does not match the expected value.

Comment (jazzybluesea) [reply include/openenclave/bits/result.h]: Done.

**Comment by jazzybluesea** [comment]
bors +r

**Pull request merged**